pgzero/actor.py

```diff
@@ -73,12 +78,17 @@ def transform_anchor(ax, ay, w, h, angle):
 
 def _set_angle(actor, current_surface):
     if actor._angle % 360 == 0:
+        # No changes required for default angle.
         return current_surface
     return pygame.transform.rotate(current_surface, actor._angle)
 
 def _set_opacity(actor, current_surface):
-    alpha_img = pygame.Surface(current_surface.get_rect().size, pygame.SRCALPHA)
-    alpha_img.fill((255, 255, 255, actor._opacity))
+    if actor.opacity == DEFAULT_OPACITY:
```

Comment:
An exact comparison is not good practice for floats.

I would compare as

```
actor.opacity > 254.5/255
```

which rounds up and therefore has better behaviour in the case that opacity is the result of a calculation.

Reply:
Good catch!

I have changed the code to make an integer comparison instead by using the calculated alpha value.